### ISSUE TITLE
Syntax Errors cause panic that can't be caught in JS

### ISSUE THREAD

**Issue by developit** [fz6m/rs-module-lexer#3]
### Describe the bug

I believe this may just need errors to be cast to [napi::Error](https://docs.rs/napi/latest/napi/struct.Error.html).

Given the following example:

```js
import {parse} from 'rs-module-lexer';

try {
  parse({
    input: [{
      filename: 'x.jsx',
      code: '(``<div />)'
    }]
  });
} catch (err) {
  // never runs
}
```

Node exits with a Rust panic message:

```
node
> const{parse} = require('rs-module-lexer');
... try {
...   parse({
...     input: [{
...       filename: 'x.jsx',
...       code: '(``<div />)'
...     }]
...   });
... } catch (err) {
...   // never runs
... }
thread '<unnamed>' panicked at crates/core/src/parser.rs:52:40:
failed to parse module: Error { error: (Span { lo: BytePos(11), hi: BytePos(12), ctxt: #0 }, TS1109) }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
zsh: IOT instruction (core dumped)  node
```

### Reproduction

parse({input:[{filename:'x.jsx',code:'(``<div />)'}]})

### System Info

```Shell
System:
    OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Xeon(R) CPU @ 2.60GHz
    Memory: 31.45 GB / 62.79 GB
    Container: Yes
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    Watchman: 4.9.0 - /usr/bin/watchman
```

### Used Package Manager

yarn

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by developit** [comment]
Ah - update: this was fixed in 1b7cf70e6f99af748d900827b1002cd83cdb261f! 
@fz6m - any chance you'd be open to publishing a new release?

**Comment by fz6m** [comment]
Released in [v2.2.0](https://github.com/fz6m/rs-module-lexer/releases/tag/v2.2.0)
